phpBB/config/default/container/services_text_reparser.yml

```diff
@@ -73,6 +73,7 @@ services:
         class: phpbb\textreparser\plugins\poll_option
         arguments:
             - '@dbal.conn'
+            - '%tables.poll_options%'
```

Comment:
May I ask why add this table here if apparently is not used in the constructor, where still the POLL_OPTIONS_TABLE constant is instead?

Reply:
The situation wrt table names is a bit of a mess. The reason some plugins use constants in their code is because the table names would have to be passed to the constructor and I didn't want to have as many different constructor signatures as there are plugins. I don't know whether it's better or worse than using constants but I figure that since those constants are used in main files, they're guaranteed to remain for at least a while.

Technically, it would be possible to keep the same constructor everywhere and pass the table names via a service method call. I don't like the idea of having essential data set outside of the constructor. It only matters when instances are created manually (in tests for example), obviously Symfony's container won't randomly forget to call the relevant method before using that object.

TL;DR: there are other ways to do the same thing. They may be better, but I'm not sure and in that situation I default to whatever requires less code/complexity.